connector/forwardconnector/forward.go

```diff
@@ -0,0 +1,131 @@
+	// separately but they must not create separate objects. When the connector is shutdown
+	// it should be removed from this map so the same configuration can be recreated successfully.
+	*sharedcomponent.SharedComponents
+}
+
+// NewFactory returns a connector.Factory.
+func NewFactory() connector.Factory {
+	f := &forwardFactory{sharedcomponent.NewSharedComponents()}
+	return connector.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		connector.WithTracesToTraces(f.createTracesToTraces, component.StabilityLevelDevelopment),
+		connector.WithMetricsToMetrics(f.createMetricsToMetrics, component.StabilityLevelDevelopment),
+		connector.WithLogsToLogs(f.createLogsToLogs, component.StabilityLevelDevelopment),
+	)
+}
+
+// createDefaultConfig creates the default configuration.
+func createDefaultConfig() component.Config {
+	return &struct{}{}
+}
+
+// createTracesToTraces creates a trace receiver based on provided config.
+func (f *forwardFactory) createTracesToTraces(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Traces,
+) (connector.Traces, error) {
+	comp := f.GetOrAdd(cfg, func() component.Component {
+		return &forward{}
+	})
+
+	conn := comp.Unwrap().(*forward)
+	conn.Traces = nextConsumer
+	return conn, nil
+}
+
+// createMetricsToMetrics creates a metrics receiver based on provided config.
+func (f *forwardFactory) createMetricsToMetrics(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Metrics,
+) (connector.Metrics, error) {
+	comp := f.GetOrAdd(cfg, func() component.Component {
+		return &forward{}
+	})
+
+	conn := comp.Unwrap().(*forward)
+	conn.Metrics = nextConsumer
+	return conn, nil
+}
+
+// createLogsToLogs creates a log receiver based on provided config.
+func (f *forwardFactory) createLogsToLogs(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Logs,
+) (connector.Logs, error) {
+	comp := f.GetOrAdd(cfg, func() component.Component {
+		return &forward{}
+	})
+
+	conn := comp.Unwrap().(*forward)
+	conn.Logs = nextConsumer
+	return conn, nil
+}
+
+// forward is used to pass signals directly from one pipeline to another.
+// This is useful when there is a need to replicate data and process it in more
+// than one way. It can also be used to join pipelines together.
+type forward struct {
+	consumer.Traces
+	consumer.Metrics
+	consumer.Logs
+	component.StartFunc
+	component.ShutdownFunc
+}
+
+func (c *forward) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *forward) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	return c.Traces.ConsumeTraces(ctx, td)
+}
+
+func (c *forward) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	return c.Metrics.ConsumeMetrics(ctx, md)
+}
+
+func (c *forward) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	return c.Logs.ConsumeLogs(ctx, ld)
+}
```

Comment:
If I don't miss something, you don't need these 3 func overrides.

Reply:
Good point. This was an artifact of the earlier implementation (e.g. `ConsumerLogsToLogs`). Removed.